docs/next/guides/building-and-testing/modules.md

```diff
@@ -481,352 +275,822 @@ new Handsontable(container, {});
+
+#### Step 1: Import the required modules
+
+First, make sure to [import the required modules](#importing-required-modules):
 
 ```js
-// import the `registerAllPlugins()` function
-import { registerAllPlugins } from 'handsontable/registry';
+// import the `handsontable/base` module
+import Handsontable from 'handsontable/base';
 
-// register all plugins
-registerAllPlugins();
+// import Handsontable's CSS
+import 'handsontable/dist/handsontable.full.css';
 ```
 
-## Importing translations
+#### Step 2: Import the translations' registering method
+
+Import the `registerLanguageDictionary()` method that lets you register your translation:
+
+```js
+import Handsontable from 'handsontable/base';
+import 'handsontable/dist/handsontable.full.css';
+
+// import the `registerLanguageDictionary()` method
+import { registerLanguageDictionary } from 'handsontable/i18n';
+```
 
-Importing translations works slightly different than in case of other elements. Let's try adding the `pl-PL` language.
+#### Step 3: Import your translation module
 
-Start with importing the base and the language code:
+Import a translation module of your choice. For example:
 
 ```js
-// remember to have the base imported
 import Handsontable from 'handsontable/base';
+import 'handsontable/dist/handsontable.full.css';
 
-// import the language code and interface to register language
+// import the `plPL` module
 import { registerLanguageDictionary, plPL } from 'handsontable/i18n';
 ```
 
-Afterwards you need to register it:
+#### Step 4: Register your translation
+
+Register your translation, using the `registerLanguageDictionary()` method. For example:
 
 ```js
-registerLanguageDictionary(plPL.languageCode, plPL);
+import Handsontable from 'handsontable/base';
+import 'handsontable/dist/handsontable.full.css';
+import { registerLanguageDictionary, plPL } from 'handsontable/i18n';
 
-// or if the language dictionary object contains the "languageCode" property,
-// the registration can be simplified to something like this
-registerLanguageDictionary(plPL);
+// register the `plPL` translation
+registerLanguageDictionary(plPL.languageCode, plPL);
 ```
 
-Now, you can use newly registered translation. The full example looks like this:
+#### Step 5: Use your translation
+
+Now, you can start using your translation, by setting the [`language`](@/api/options.md#language) [configuration option](@/guides/getting-started/setting-options.md):
 
 ```js
 import Handsontable from 'handsontable/base';
+import 'handsontable/dist/handsontable.full.css';
 import { registerLanguageDictionary, plPL } from 'handsontable/i18n';
 
-registerLanguageDictionary(plPL);
+registerLanguageDictionary(plPL.languageCode, plPL);
 
-// use the locales
+// set the `language` configuration option
 new Handsontable(container, {
   language: 'pl-PL',
-// rest of the settings
 });
 ```
 
-And that is all! You can use the PL-pl translation!
+:::tip
+#### Optimizing Moment.js locales
```

Comment:
The `moment.js` is not only used within the Date cell type but also as a sort comparator in the _ColumnSorting_ plugin and as condition functions (e.g. "After", "Between" ...) for date column types in the _Filters_ plugin. 

Maybe it would be some idea to move it to the new section about performance or optimization and link to it 🤔

Reply:
If I get it right, Moment.js is included in `handsontable/base`, right? This means that everone who cares about Handsontable's impact on their app should optimize Moment.js locales, no matter if they use the `date` cell type or not.

How about we add it to the **Performance** page then? (it follows the structure I suggested in [this comment](https://github.com/handsontable/handsontable/pull/9059#discussion_r787667657))
- Optimization
    - Batch operations
    - Performance
      - Optimizing Moment.js
    - Packages
    - Modules

Unless we just don't want to mention

Reply:
> If I get it right, Moment.js is included in handsontable/base, right?

Good point. The lib is used in the Core (so including `handsontable/base`) for checking if the license key is valid.

Reply:
My suggestion then is to limit the "Performance" page to run-time performance.

We can create a new page "Bundle size", where there will be two sections "Use modules" and "Optimizing Moment.js".

- **Optimization**
   - **Batch operations**
   - **Performance**
   - **Bundle size**